pakyow-core/lib/pakyow/core/router.rb

```diff
@@ -196,6 +196,15 @@ class Router
 
     # @api private
     def initialize(context)
+      if router = context.previous_router_instance
+        # copy state from previous router to the new one
+        # this is so that state is shared when rerouting from one router to another one
```

Comment:
I agree that it is better in terms of understanding the application from the point of the user, and maybe if we can encapsulate it in a method it would be better if we come up with a different way of doing it later.

Maybe something like 

```ruby
router.pass_context(other_router)
#or
router.hand_off(other_router)
```